File: codigos/TAD/exercicio_biblioteca.py
class Livro:

    def __init__(self, id, titulo):
        self.id = id
        self.titulo = titulo
        self.disponivel = True

    def emprestar(self):
        if self.disponivel:
            self.disponivel = False
            print("Livro emprestado com sucesso.")
        else:
            print("Livro indisponível.")

    def devolver(self):
        self.disponivel = True
        print("Livro devolvido.")

    def mostrar(self):
        print(f"ID: {self.id}")
        print(f"Título: {self.titulo}")
        print(f"Disponível: {self.disponivel}")


livro = Livro(1, "Estruturas de Dados")

livro.mostrar()

print()

livro.emprestar()

print()

livro.mostrar()

print()

livro.devolver()

print()

livro.mostrar()

File: codigos/TAD/exercicio_biblioteca.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

typedef struct {
    int id;
    char titulo[100];
    bool disponivel;
} Livro;

void emprestar(Livro *livro) {
    if (livro->disponivel) {
        livro->disponivel = false;
        printf("Livro emprestado com sucesso.\n");
    } else {
        printf("Livro indisponivel.\n");
    }
}

void devolver(Livro *livro) {
    livro->disponivel = true;
    printf("Livro devolvido.\n");
}

void mostrarLivro(Livro livro) {
    printf("ID: %d\n", livro.id);
    printf("Titulo: %s\n", livro.titulo);
    printf("Disponivel: %s\n",
           livro.disponivel ? "Sim" : "Nao");
}

int main() {

    Livro livro;

    livro.id = 1;
    strcpy(livro.titulo, "Estruturas de Dados");
    livro.disponivel = true;

    mostrarLivro(livro);

    printf("\n");

    emprestar(&livro);

    printf("\n");

    mostrarLivro(livro);

    printf("\n");

    devolver(&livro);

    printf("\n");

    mostrarLivro(livro);

    return 0;
}